### ISSUE TITLE
Issue when upgrading from V2.4.0 to V3.2.0

### ISSUE THREAD

**Issue by cname87** [anttiviljami/openapi-backend#81]
All was working well with V2.4.0 for the longest time - thanks.  I upgraded to V3.2.0 and now my url is not found by the OpenAPIBackend instance, i.e. the correct handler is not found and it goes to the notFound handler.  By trying different versions I traced back the issue as being introduced between v3.0.0 and v3.0.1.  I noticed you added an extra test in v3.0.1 in router.ts:

// make sure request path matches apiRoot
    if (!req.path.startsWith(this.apiRoot)) {
      return undefined;
    } 

I use Express and set my apiRoot ('api-v1') as a mountpath to route api requests which means req.path will not include the apiRoot (as per Express docs: "When called from a middleware, the mount point is not included in req.path. See app.use() for more details.").

Can you review and let me know your thoughts please?

**Comment by anttiviljami** [comment]
Hi @cname87 !

Unfortunately the pre-3.0.0 behaviour, which your code was relying on was unintended. See bug report: #44

The way I see it you have two possible solutions:

1. Just remove apiRoot from the OpenAPIBackend constructor opts
2. Supply full path to `.handleRequest()`

Honestly I would just go for option 1 to make the code look nicer. This is of course unless you're using apiRoot elsewhere in your app logic.

If you want to go for option 2, you have a couple of ways to do this. Easiest would probably be doing something like this:

```javascript
// https://expressjs.com/en/api.html#req.originalUrl
app.use((req, res) => api.handleRequest({ ...req, path: req.originalUrl }, req, res));
```

Or even:

```javascript
// https://expressjs.com/en/api.html#req.baseUrl
app.use((req, res) => api.handleRequest({ ...req, path: req.baseUrl + req.path }, req, res));
```

Sorry about the inconvenience!

**Comment by cname87** [comment]
Thanks for the reply.  Yes I see the logic of your change and have made an easy fix and upgraded to 3.4.0 and all is working.  Thanks!
